Time Base API test introduced new format warnings

**Describe the bug**
New warnings noted when building for RTEMS:
```
/home/joe/code/cfecfs/github/osal/src/tests/time-base-api-test/time-base-api-test.c:110:5: error: format '%d' expects argument of type 'int', but argument 4 has type 'int32' [-Werror=format=]
     snprintf(overMaxTimeBase, 12, "TimeBase%d", TimeBaseNum);
     ^
```

**To Reproduce**
Build code using `i686-rtems4.11` toolchain.

**Expected behavior**
Should build cleanly.

**System observed on:**
Cross compile for `i686-rtems4.11` on Ubuntu 20.04 host.

**Additional context**
Needs cast whenever fixed-width types are used with printf.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/src/tests/time-base-api-test/time-base-api-test.c
+++ b/src/tests/time-base-api-test/time-base-api-test.c
@@ -103,11 +103,11 @@ void TestTimeBaseApi(void)
     /* Checking OS_MAX_TIMEBASES + 1 */
     for ( int i = 0; i < OS_MAX_TIMEBASES; i++ )
     {
-        snprintf(TimeBaseIter[i], 12, "TimeBase%d", i);
+        snprintf(TimeBaseIter[i], sizeof(TimeBaseIter[i]), "TimeBase%d", i);
         tbc_results[i] = OS_TimeBaseCreate(&tb_id[i], TimeBaseIter[i], 0);
     }
     TimeBaseNum = OS_MAX_TIMEBASES+1;
-    snprintf(overMaxTimeBase, 12, "TimeBase%d", TimeBaseNum);
+    snprintf(overMaxTimeBase, sizeof(overMaxTimeBase), "TimeBase%d", (int)TimeBaseNum);
     expected = OS_ERR_NO_FREE_IDS;
     actual= OS_TimeBaseCreate(&time_base_id, "overMaxTimeBase", 0);
     UtAssert_True(actual == expected, "OS_TimeBaseCreate() (%ld) == OS_ERR_NO_FREE_IDS", (long)actual);

